generators/Generators.fs

```diff
@@ -267,6 +267,96 @@ type RailFenceCipher() =
+        // one identifier may be empty if we only checking created object
+        // we need to filter out this empty line
+        base.RenderArrange canonicalDataCase
+        |> List.choose (
+            function
+            | "" -> None
+            | v -> Some v
+        )
+
+    override this.RenderValueWithoutIdentifier (canonicalDataCase, key, value) = 
+        match key with
+        | "robot" ->
+            let input = value :?> JToken
+            this.DefineRobot ("direction" |> input.SelectToken) ("position" |> input.SelectToken)
+        | "expected" -> 
+            // here we may need to render full robot object
+            // or just coordinate/direction values
+            let input = (value :?> JToken)
+            let direction, position = this.GetRobotProperties input
+            match isNull(direction), isNull(position) with
+            | true, false ->
+                this.RenderCoords position
+            | false, true ->
+                this.RenderDirection direction
+            | false, false ->
+                this.DefineRobot ("direction" |> input.SelectToken) ("position" |> input.SelectToken)
+            | true, true -> 
+                ""
+        | _ -> ""
+
+    override this.RenderValueWithIdentifier (canonicalDataCase, key, value) = 
+        match key with 
+        | "property" -> 
+            let action = (value :?> string)
+            match action with 
+            | "instructions" -> 
+                 sprintf "let %s = simulate robot %s" this.ResultIdentifierName (canonicalDataCase.Properties.["instructions"] |> formatValue)
+            | "create" -> 
+                ""
+            | _ -> 
+                 sprintf "let %s = %s robot" this.ResultIdentifierName action
+            
+        | _ -> 
+            base.RenderValueWithIdentifier (canonicalDataCase, key, value)
+
+    override this.RenderSut canonicalDataCase = 
+        match canonicalDataCase.Property with
+        | "create" -> 
+            "robot"
+        | _ -> 
+            // depends on expected value we may need to 
+            // check whole robot or just one of its' properties
+            let direction, position = this.GetRobotProperties (canonicalDataCase.Properties.["expected"] :?> JToken)
+            match isNull(direction), isNull(position) with
+            | true, false -> 
+                sprintf "%s.coordinate" this.ResultIdentifierName
+            | false, true ->
+                sprintf "%s.bearing" this.ResultIdentifierName
+            | true, true -> 
+                this.ResultIdentifierName
+            | false, false -> 
+                match canonicalDataCase.Property with
+                | "create" -> "robot"
+                | _ -> this.ResultIdentifierName
+    
+    override this.RenderTestMethodName canonicalDataCase =
+        // avoid duplicated method names
+        sprintf "%s - %s" canonicalDataCase.Property (canonicalDataCase.Description |> String.upperCaseFirst)
```

Comment:
There is another option to avoid duplicated method names, see [this example](https://github.com/exercism/fsharp/blob/master/generators/Generators.fs#L145). However, I've looked at the canonical data, and it looks like that would lead to incredibly long method names, so I'm fine with leaving this as is.